Detections/ASimNetworkSessions/PossibleBeaconingActivity.yaml

```diff
@@ -0,0 +1,65 @@
+id: fcb9d75c-c3c1-4910-8697-f136bfef2363
+name: Potential beaconing activity
+description: |
+  'Identifies beaconing patterns from Network traffic logs based on recurrent timedelta patterns. 
+  The query leverages various KQL functions to calculate time deltas and then compares it with total events observed in a day to find percentage of beaconing. 
+  This outbound beaconing pattern to untrusted public networks should be investigated for any malware callbacks or data exfiltration attempts.
+  Reference Blog:
+  http://www.austintaylor.io/detect/beaconing/intrusion/detection/system/command/control/flare/elastic/stack/2017/06/10/detect-beaconing-with-flare-elasticsearch-and-intrusion-detection-systems/'
+severity: Low
+requiredDataConnectors: []
+queryFrequency: 2d
+queryPeriod: 1d
+triggerOperator: gt
+triggerThreshold: 0
+tactics:
+  - CommandAndControl
+relevantTechniques:
+  - T1071
+  - T1571
+tags:
+  - Id: f0be259a-34ac-4946-aa15-ca2b115d5feb
+    version: 1.0.0
+  - Schema: ASIMNetworkSession
+    SchemaVersion: 0.2.0
+
+query: |
+  let querystarttime = 2d;
+  let queryendtime = 1d;
+  let TimeDeltaThreshold = 10;
+  let TotalEventsThreshold = 15;
+  let PercentBeaconThreshold = 80;
+  imNetworkSession(starttime=querystarttime, endtime=queryendtime)
+  | where not(ipv4_is_private(DstIpAddr))
+  | project TimeGenerated, DvcHostname, SrcUserId, SrcIpAddr, SrcPortNumber, DstIpAddr, DstPortNumber, DstBytes, SrcBytes
```

Comment:
Not all data may be available from each source. It would be important to handle that correctly. In most cases just not alert. You might want to actually prequalify if a field is important (i.e. where DstPortNumber > 0 and DstIpAddr != "")

Reply:
I meant in generally. Something we need to think of in each query. If it is fine here, all good.